lib/kazan/watcher.ex

```diff
@@ -0,0 +1,164 @@
+defmodule Kazan.Watcher do
+  @moduledoc """
+
+  Watches for changes on a resource.  This works by creating a HTTPoison ASync request.
+  The request will eventually timeout, however the Watcher handles recreating the request
+  when this occurs and requesting new events that have occurred since the last *resource_version*
+  received.
+
+  To use:
+
+  1. Create a request in the normal way, but setting `watch=true` e.g.
+  ```
+  request = Kazan.Apis.CoreV1.list_namespace!(watch: "true")
+  ```
+  2. Start a watcher using the request, and passing the initial resource version
+  and a pid to which to send received messages.  The initial *resource_version* can be
+  obtained by doing a normal, non-watch request to the same endpoint.
```

Comment:
Is it possible for the Watcher to support not passing in a `resource_version`?  If someone is non-specific then maybe it could run an initial `GET` to figure out the latest version, and then run a watch based on that...